### ISSUE TITLE
KeyError when updating a user

### ISSUE THREAD

**Issue by Plyrolith** [cgwire/zou#470]
When updating a user, I'm getting a key error for 'departments'.

We're currently on 14.4, this has been happening since 13.57. 
(This is the actual error leading to #459).

I've successfully run `zou upgrade-db` on every upgrade. Could this still be an issue with our database?

Thanks!

```log
zou-app      | [2022-05-20 08:23:41 +0000] [9] [ERROR] Error handling request //data/persons/621b5743-56e8-4eff-b316-b8a58f0b9be0
zou-app      | Traceback (most recent call last):
zou-app      |   File "/usr/local/lib/python3.7/site-packages/gunicorn/workers/base_async.py", line 55, in handle
zou-app      |     self.handle_request(listener_name, req, client, addr)
zou-app      |   File "/usr/local/lib/python3.7/site-packages/gunicorn/workers/ggevent.py", line 143, in handle_request
zou-app      |     super().handle_request(listener_name, req, sock, addr)
zou-app      |   File "/usr/local/lib/python3.7/site-packages/gunicorn/workers/base_async.py", line 106, in handle_request
zou-app      |     respiter = self.wsgi(environ, resp.start_response)
zou-app      |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 2463, in __call__
zou-app      |     return self.wsgi_app(environ, start_response)
zou-app      |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 2449, in wsgi_app
zou-app      |     response = self.handle_exception(e)
zou-app      |   File "/usr/local/lib/python3.7/site-packages/flask_restful/__init__.py", line 272, in error_router
zou-app      |     return original_handler(e)
zou-app      |   File "/usr/local/lib/python3.7/site-packages/flask_restful/__init__.py", line 272, in error_router
zou-app      |     return original_handler(e)
zou-app      |   File "/usr/local/lib/python3.7/site-packages/flask_restful/__init__.py", line 272, in error_router
zou-app      |     return original_handler(e)
zou-app      |   [Previous line repeated 17 more times]
zou-app      |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1866, in handle_exception
zou-app      |     reraise(exc_type, exc_value, tb)
zou-app      |   File "/usr/local/lib/python3.7/site-packages/flask/_compat.py", line 38, in reraise
zou-app      |     raise value.with_traceback(tb)
zou-app      |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 2446, in wsgi_app
zou-app      |     response = self.full_dispatch_request()
zou-app      |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1951, in full_dispatch_request
zou-app      |     rv = self.handle_user_exception(e)
zou-app      |   File "/usr/local/lib/python3.7/site-packages/flask_restful/__init__.py", line 272, in error_router
zou-app      |     return original_handler(e)
zou-app      |   File "/usr/local/lib/python3.7/site-packages/flask_restful/__init__.py", line 272, in error_router
zou-app      |     return original_handler(e)
zou-app      |   File "/usr/local/lib/python3.7/site-packages/flask_restful/__init__.py", line 272, in error_router
zou-app      |     return original_handler(e)
zou-app      |   [Previous line repeated 17 more times]
zou-app      |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1820, in handle_user_exception
zou-app      |     reraise(exc_type, exc_value, tb)
zou-app      |   File "/usr/local/lib/python3.7/site-packages/flask/_compat.py", line 38, in reraise
zou-app      |     raise value.with_traceback(tb)
zou-app      |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1949, in full_dispatch_request
zou-app      |     rv = self.dispatch_request()
zou-app      |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1935, in dispatch_request
zou-app      |     return self.view_functions[rule.endpoint](**req.view_args)
zou-app      |   File "/usr/local/lib/python3.7/site-packages/flask_restful/__init__.py", line 468, in wrapper
zou-app      |     resp = resource(*args, **kwargs)
zou-app      |   File "/usr/local/lib/python3.7/site-packages/flask/views.py", line 89, in view
zou-app      |     return self.dispatch_request(*args, **kwargs)
zou-app      |   File "/usr/local/lib/python3.7/site-packages/flask_restful/__init__.py", line 583, in dispatch_request
zou-app      |     resp = meth(*args, **kwargs)
zou-app      |   File "/usr/local/lib/python3.7/site-packages/flask_jwt_extended/view_decorators.py", line 108, in wrapper
zou-app      |     return fn(*args, **kwargs)
zou-app      |   File "/usr/local/lib/python3.7/site-packages/zou/app/blueprints/crud/base.py", line 264, in get
zou-app      |     result = self.serialize_instance(instance)
zou-app      |   File "/usr/local/lib/python3.7/site-packages/zou/app/blueprints/crud/person.py", line 82, in serialize_instance
zou-app      |     return instance.present_minimal()
zou-app      |   File "/usr/local/lib/python3.7/site-packages/zou/app/models/person.py", line 88, in present_minimal
zou-app      |     "departments": data["departments"],
zou-app      | KeyError: 'departments'
```

**Comment by EvanBldy** [comment]
Hi @Plyrolith, can you open psql to check that the department link with person exist ?
`sudo -u postgres psql`
In psql type : 
```
postgres=# \c zoudb
You are now connected to database "zoudb" as user "postgres".
zoudb=# \d+ person
```
```
                                                          Table "public.person"
              Column              |            Type             | Collation | Nullable | Default | Storage  | Stats target | Description 
----------------------------------+-----------------------------+-----------+----------+---------+----------+--------------+-------------
 id                               | uuid                        |           | not null |         | plain    |              | 
 created_at                       | timestamp without time zone |           |          |         | plain    |              | 
 updated_at                       | timestamp without time zone |           |          |         | plain    |              | 
 first_name                       | character varying(80)       |           | not null |         | extended |              | 
 last_name                        | character varying(80)       |           | not null |         | extended |              | 
 email                            | character varying(255)      |           |          |         | extended |              | 
 phone                            | character varying(30)       |           |          |         | extended |              | 
 active                           | boolean                     |           |          |         | plain    |              | 
 last_presence                    | date                        |           |          |         | plain    |              | 
 password                         | bytea                       |           |          |         | extended |              | 
 desktop_login                    | character varying(80)       |           |          |         | extended |              | 
 shotgun_id                       | integer                     |           |          |         | plain    |              | 
 timezone                         | character varying(50)       |           |          |         | extended |              | 
 locale                           | character varying(10)       |           |          |         | extended |              | 
 data                             | jsonb                       |           |          |         | extended |              | 
 role                             | character varying(30)       |           |          |         | extended |              | 
 has_avatar                       | boolean                     |           |          |         | plain    |              | 
 notifications_enabled            | boolean                     |           |          |         | plain    |              | 
 notifications_slack_enabled      | boolean                     |           |          |         | plain    |              | 
 notifications_slack_userid       | character varying(60)       |           |          |         | extended |              | 
 notifications_mattermost_enabled | boolean                     |           |          |         | plain    |              | 
 notifications_mattermost_userid  | character varying(60)       |           |          |         | extended |              | 
 notifications_discord_enabled    | boolean                     |           |          |         | plain    |              | 
 notifications_discord_userid     | character varying(60)       |           |          |         | extended |              | 
Indexes:
    "person_pkey" PRIMARY KEY, btree (id)
    "person_email_key" UNIQUE CONSTRAINT, btree (email)
    "person_shotgun_id_key" UNIQUE CONSTRAINT, btree (shotgun_id)
Referenced by:
    TABLE "api_event" CONSTRAINT "api_event_user_id_fkey" FOREIGN KEY (user_id) REFERENCES person(id)
    TABLE "assignations" CONSTRAINT "assignations_person_fkey" FOREIGN KEY (person) REFERENCES person(id)
    TABLE "comment_acknoledgments" CONSTRAINT "comment_acknoledgments_person_fkey" FOREIGN KEY (person) REFERENCES person(id)
    TABLE "comment_mentions" CONSTRAINT "comment_mentions_person_fkey" FOREIGN KEY (person) REFERENCES person(id)
    TABLE "comment" CONSTRAINT "comment_person_id_fkey" FOREIGN KEY (person_id) REFERENCES person(id)
    TABLE "day_off" CONSTRAINT "day_off_person_id_fkey" FOREIGN KEY (person_id) REFERENCES person(id)
    TABLE "department_link" CONSTRAINT "department_link_person_id_fkey" FOREIGN KEY (person_id) REFERENCES person(id)
    TABLE "desktop_login_log" CONSTRAINT "desktop_login_log_person_id_fkey" FOREIGN KEY (person_id) REFERENCES person(id)
    TABLE "entity_version" CONSTRAINT "entity_version_person_id_fkey" FOREIGN KEY (person_id) REFERENCES person(id)
    TABLE "login_log" CONSTRAINT "login_log_person_id_fkey" FOREIGN KEY (person_id) REFERENCES person(id)
    TABLE "news" CONSTRAINT "news_author_id_fkey" FOREIGN KEY (author_id) REFERENCES person(id)
    TABLE "notification" CONSTRAINT "notification_author_id_fkey" FOREIGN KEY (author_id) REFERENCES person(id)
    TABLE "notification" CONSTRAINT "notification_person_id_fkey" FOREIGN KEY (person_id) REFERENCES person(id)
    TABLE "output_file" CONSTRAINT "output_file_person_id_fkey" FOREIGN KEY (person_id) REFERENCES person(id)
    TABLE "preview_file" CONSTRAINT "preview_file_person_id_fkey" FOREIGN KEY (person_id) REFERENCES person(id)
    TABLE "project_person_link" CONSTRAINT "project_person_link_person_id_fkey" FOREIGN KEY (person_id) REFERENCES person(id)
    TABLE "search_filter" CONSTRAINT "search_filter_person_id_fkey" FOREIGN KEY (person_id) REFERENCES person(id)
    TABLE "subscription" CONSTRAINT "subscription_person_id_fkey" FOREIGN KEY (person_id) REFERENCES person(id)
    TABLE "task" CONSTRAINT "task_assigner_id_fkey" FOREIGN KEY (assigner_id) REFERENCES person(id)
    TABLE "time_spent" CONSTRAINT "time_spent_person_id_fkey" FOREIGN KEY (person_id) REFERENCES person(id)
    TABLE "working_file" CONSTRAINT "working_file_person_id_fkey" FOREIGN KEY (person_id) REFERENCES person(id)
Access method: heap

```
Check that the line `TABLE "department_link" CONSTRAINT "department_link_person_id_fkey" FOREIGN KEY (person_id) REFERENCES person(id)` is here

**Comment by EvanBldy** [comment]
Hi @Plyrolith,
Finally I have found the bug, the fix is here : https://github.com/cgwire/zou/pull/472
It will be in the next release !
Can you close both issues when you've tested it's fixed?
Thanks for the report !
Have a nice day :)

**Comment by Plyrolith** [comment]
Hey Evan,
Fix #474  tested & confirmed!
Thank you so much!
